zerver/actions/message_edit.py

```diff
@@ -901,6 +905,12 @@ def check_update_message(
     if not user_profile.realm.allow_message_editing:
         raise JsonableError(_("Your organization has turned off message editing"))
 
+    stream = None
+    if message.is_stream_message():
+        stream = Stream.objects.get(id=message.recipient.type_id)
+        if stream.deactivated:
+            raise JsonableError(_("Stream does not exist"))
```

Comment:
Have not used `access_stream_by_id` here because `access_stream_by_id` raises error for the streams which are private and user was subscribed to before but not now. We allow user to edit message content in such case. And since we have stream object here I changed the calls to `access_stream_by_id` below to `access_stream_common` such that we do not want to fetch stream object again. 

Though I am not sure about whether there can be case where the id in `message.recipient.type_id` field is incorrect, i.e. not of a stream belonging to this realm. I checked in couple of places, in reactions and uploads code, we do not use try-except block while getting stream object from `message.recipient.type_id`.